src/main/java/com/epam/asset/tracking/service/UserServiceImpl.java

```diff
@@ -29,4 +49,33 @@ public UserDetails loadUserByUsername(String userName) throws UsernameNotFoundEx
     return Arrays.asList(new SimpleGrantedAuthority("ROLE_BUSINESS_PROVIDER"));
   }
 
+  @Override
+  public BusinessProvider generateNewPassword(String username) throws InvalidUserException {
+    PasswordGenerator passwordGenerator = new PasswordGenerator();
+
+    CharacterRule characterRule = new CharacterRule(EnglishCharacterData.Alphabetical);
+
+    newPassword = passwordGenerator.generatePassword(8, characterRule);
```

Comment:
The reason is because, we need to send to email so before that is being encoded, so I am using the instance variable to send the unencoded password through email